src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,137 @@
+package duke;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import exception.EmptyDescriptionException;
+import exception.EmptyTimeException;
+import exception.InvalidCommandException;
+import exception.InvalidIndexException;
+
+/**
+ * Represents Duke, a chat bot that allows tasks management (add, delete, mark as done).
+ * @author Lucia Tirta Gunawan
+ * @author A0200718N
+ */
+public class Duke {
+    /** Storage to load and save tasks. */
+    private Storage storage;
+    /** List of tasks saved in Duke. */
+    private TaskList taskList;
+    /** User interface to display output from Duke. */
+    private Ui ui;
+
+    /**
+     * Constructor for Duke.
+     * @param filePath the file path of the saved tasks.
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            taskList = new TaskList(storage.loadData());
+        } catch (FileNotFoundException e) {
+            taskList = new TaskList();
+        }
+    }
+
+    /**
+     * Gets welcome message.
+     * @return welcome message from Duke.
+     */
+    public String getWelcomeMessage() {
+        return ui.gettWelcomeMessage();
+    }
+
+    /**
+     * Gets response from Duke after giving command
+     * @param input the user input
+     * @return the String response from Duke
+     */
+    public String getResponse(String input) {
+//        ui.printWelcome();
+        String[] inputs = input.split("\\s+", 2);
+        try {
+            Commands command = Parser.parse(input);
+            if (command.equals(Commands.BYE)) {
+                try {
```

Comment:
There are a number of try catch blocks within each if block, perhaps it might be possible to use a wider try catch block with more catch blocks to deal with multiple exceptions, or handle them simultaneously since the exceptions are handled similarly.